test: add enable_admin_tests and require_admin() to Python test framework

ldorau opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmdk/4803)
<!-- Reviewable:end -->

pbalcer commented on src/test/unittest/utils.py:
```diff
@@ -45,6 +50,33 @@ def wrapped(tc):
+            futils.fail('{}: FAILED: admin tests are enabled but the user'
+                        ' does not have administrative privileges'.format(tc))
+    else:
+        print('{}: SKIP: admin tests are disabled in config'
```
`futils.skip("admin tests ...")` ?

I checked and the "SKIP" message is already generated in RUNTESTS.py:67.

ldorau replied:
If `futils.skip()` is used, the message is not printed out ...

pbalcer replied:
Then that should be fixed instead of adding a manual print.
Are you testing with `RUNTESTS.py` or by manually running `TESTS.py`, because I can see that the message is *not* printed in the `TESTS.py` variant.

pbalcer commented:
Have you seen #4722 ? I like your approach better, but you can see how Leszek did it.
Should we close that PR?

ldorau commented:
> Have you seen #4722 ? I like your approach better, but you can see how Leszek did it.
> Should we close that PR?

I have not seen it before :-) I will check the Leszek's PR - it should be better than mine ;-)

pbalcer commented:
I like yours better because, although arguably slower, it's self-contained.

ldorau commented:
>   * it CAN NOT use config and futils (which are Test Framework utilities)

@janekmi I am afraid it has to use config (@llugin used it in his commit https://github.com/pmem/pmdk/pull/4722/commits/524ff9b5686b965712a2526476eb05ceaa57028a)

ldorau commented:
> I just realized... we can not implement this functionality in this way. Sorry, @ldorau one more time.
> 
> I know it is not yet well documented but Python TF aims to keep some entities separately.
> I'm not yet on top of this but e.g. a test is an entity of itself and has no idea about Test Framework and its configuration [...]

I have incorporated https://github.com/pmem/pmdk/pull/4722 and corrected it a bit. I think it the right approach.

ldorau commented on src/test/RUNTESTS.py:
```diff
@@ -25,6 +26,9 @@ class TestRunner:
     def __init__(self, config, testcases):
         self.testcases = testcases
         self.config = config
+        self._check_sudo()
```
Done

ldorau commented:
It depends on https://github.com/pmem/pmdk/pull/4806
